stdlib/Serialization/src/Serialization.jl

```diff
@@ -1201,26 +1201,37 @@ function deserialize(s::AbstractSerializer, ::Type{CodeInfo})
     if pre_12
         ci.slotflags = deserialize(s)
     else
-        ci.method_for_inference_limit_heuristics = deserialize(s)
+        if format_version(s) <= 26
+            ci.method_for_inference_limit_heuristics = deserialize(s)
+        end
         ci.linetable = deserialize(s)
     end
     ci.slotnames = deserialize(s)
     if !pre_12
         ci.slotflags = deserialize(s)
         ci.slottypes = deserialize(s)
-        ci.rettype = deserialize(s)
-        ci.parent = deserialize(s)
-        world_or_edges = deserialize(s)
-        pre_13 = isa(world_or_edges, Integer)
-        if pre_13
-            ci.min_world = world_or_edges
+        if format_version(s) <= 26
+            deserialize(s) # rettype
+            deserialize(s) # parent
+            world_or_edges = deserialize(s)
+            pre_13 = isa(world_or_edges, Integer)
+            if pre_13
+                ci.min_world = world_or_edges
+            else
+                ci.edges = world_or_edges
+                ci.min_world = reinterpret(UInt, deserialize(s))
+                ci.max_world = reinterpret(UInt, deserialize(s))
+            end
         else
-            ci.edges = world_or_edges
+            ci.method_for_inference_limit_heuristics = deserialize(s)
+            ci.edges = deserialize(s)
             ci.min_world = reinterpret(UInt, deserialize(s))
             ci.max_world = reinterpret(UInt, deserialize(s))
         end
     end
-    ci.inferred = deserialize(s)
+    if format_version(s) <= 26
+        deserialize(s) # inferred
+    end
     if format_version(s) < 22
```

Comment:
This is quite buggy before, based on my experience in https://github.com/JuliaLang/julia/pull/52415/files#diff-0fbb91f060d958d93fcf0a101ade4d87b365f3664a409276c591d7faf77acb67. Attempting to fix this just by comparison though, so hopefully I am manipulating this right (I think the whole mess with reinterpreting an Integer as a UInt there is because the field was once `Union{Int,UInt}` actually, when taken across all of the old versions:

```suggestion
        if format_version(s) <= 26
            deserialize(s) # rettype
            deserialize(s) # parent
            world_or_edges = deserialize(s)
            pre_13 = isa(world_or_edges, Integer)
            if !pre_13
                ci.edges = world_or_edges
                world_or_edges = deserialize(s)::Integer
            end
            ci.min_world = reinterpret(UInt, world_or_edges)
            ci.max_world = reinterpret(UInt, deserialize(s))
        else
            ci.method_for_inference_limit_heuristics = deserialize(s)
            ci.edges = deserialize(s)
            ci.min_world = deserialize(s)::UInt
            ci.max_world = deserialize(s)::UInt
        end
    end
    if format_version(s) <= 26
        deserialize(s)::Bool # inferred
    end
    if format_version(s) < 22
```

Reply:
I'm not sure what the change is that you're asking for here.